src/components/main/HomeScreen_alt.js

```diff
@@ -0,0 +1,346 @@
+import React from 'react'
+
+import {
+  TouchableOpacity,
+  TouchableWithoutFeedback,
+  StyleSheet,
+  Dimensions,
+  Text,
+  Keyboard,
+  View,
+  ImageBackground
+} from 'react-native'
+
+import { Linking } from 'expo';
+
+import { Input, normalize } from 'react-native-elements'
+
+import { NavigationEvents } from 'react-navigation'
+
+import { FontAwesome, Ionicons } from '@expo/vector-icons'
+
+import COLORS from '../../assets/Colors.js'
+
+// import Footer from './Footer.js'
+
+import MenuBar from './MenuBar.js'
+
+import Auth from '@aws-amplify/auth'
+
+import i18n from 'i18n-js'
+
+import * as CONSTANTS from '../utilities/Constants.js'
+
+const background_image = require('../../assets/background/home/bg_home.png')
+
+export default class HomeScreen extends React.Component {
+  state = {
+    isAuthenticated: false,
+    flight: '',
+    error: false,
+  }
+
+  // load background
+  constructor(props) {
+    super(props)
+    this.selectBackground()
+  }
+
+  onChangeText(key, value) {
+    this.setState({
+      [key]: value
+    })
+  }
+
+  // selects a random background to be loaded
+  selectBackground() {
+    this.background = background_image
+  }
+
+  // Opens up email to bug-report
+  handleBugReports = () => {
+    Linking.openURL('mailto:bug-report@refloresta.app?body=' + CONSTANTS.BUG_REPORT);
+  }
+
+  // Checks if a user is logged in
+  async checkAuth() {
+    await Auth.currentAuthenticatedUser({ bypassCache: true })
+      .then(() => {
+        console.log('A user is logged in')
+        this.setState({ isAuthenticated: true })
+      })
+      .catch(err => {
+        console.log('Nobody is logged in')
+        this.setState({ isAuthenticated: false })
+      })
+  }
+
+  // Check if the inputed flight number is valid
+  checkNum() {
+    let spaceBuffer = this.state.flight.replace(/\s+/g, '');
+
+    this.setState({
+      flight: spaceBuffer
+    })
+
+    let charsIata = spaceBuffer.slice(0, 2).toUpperCase();
+    let charsIcao = spaceBuffer.slice(0, 3).toUpperCase();
+
+    console.log(charsIata);
+    console.log(charsIcao);
+
+    let numsIata = spaceBuffer.slice(2);
+    let numsIcao = spaceBuffer.slice(3);
+
+    console.log(numsIata);
+    console.log(numsIcao);
+
```

Comment:
no error is ever thrown, we are taken to checkout with flight no matter what here

Reply:
errors are thrown in iataCall and icaoCall

Reply:
can you elaborate on this?